### PR TITLE
Disambiguate names of service methods in grpc output

### PR DESCRIPTION
**Pull request by sigurdm**
Fixes: https://github.com/dart-lang/protobuf/issues/485

### CODE REVIEW

File: protoc_plugin/lib/names.dart [hunk 1]
```diff
@@ -506,6 +506,23 @@ final List<String> forbiddenExtensionNames = <String>[
   ..._generatedMessageNames
 ];
 
+final List<String> serviceReservedMemberNames = <String>[
+  ..._dartReservedWords,
+  ..._serviceNames,
+];
+
+final List<String> _serviceNames = <String>[
```
Comment (nichite) [inline protoc_plugin/lib/names.dart:514]: Can this be `const`? Also, do we need the explicit type on the left-hand side?

File: protoc_plugin/test/goldens/grpc_service.pbgrpc [hunk 2]
```diff
@@ -30,6 +30,18 @@ class TestClient extends $grpc.Client {
       '/Test/Bidirectional',
       ($0.Input value) => value.writeToBuffer(),
       ($core.List<$core.int> value) => $0.Output.fromBuffer(value));
+  static final _$new_ = $grpc.ClientMethod<$0.Input, $0.Output>(
```
Comment (nichite) [inline protoc_plugin/test/goldens/grpc_service.pbgrpc:33]: One thing I was curious of, is it possible to have a name collision with the library import aliases? i.e. do we also need to add `r'$grpc'`, `r'$async'` etc to the reserved words?

**Review by nichite — COMMENTED** [review]
Mostly LGTM

File: protoc_plugin/test/goldens/grpc_service.pbgrpc (hunk 2 above)
Comment (sigurdm) [reply protoc_plugin/test/goldens/grpc_service.pbgrpc:33]: $ cannot occur in a protobuf identifier, (https://developers.google.com/protocol-buffers/docs/reference/proto3-spec#identifiers) - I also tried it out and the protoc compiler gave a syntax error.

I believe that is why we chose to use $ in the first place for these names.

That means we could actually remove all the reserved words with $!

I won't do that in this PR to keep the scope smaller.

File: protoc_plugin/lib/names.dart (hunk 1 above)
Comment (sigurdm) [reply protoc_plugin/lib/names.dart]: yes!

Done - also in surrounding code.

File: protoc_plugin/test/goldens/grpc_service.pbgrpc (hunk 2 above)
Comment (sigurdm) [reply protoc_plugin/test/goldens/grpc_service.pbgrpc:33]: (I removed it from those added).

**Comment by sigurdm** [comment]
I realized we also needed disambiguation for the .pbserver.dart code, and added that - this will require a bit of integration work when moving into google3.

@nichite @iinozemtsev do you know if we have any real uses of the pbserver.dart files internally? Searching for 'import.*.pbserver.dart' gives no real results. Maybe it is time to retire that code, and have the grpc code be generated by default?

**Comment by nichite** [comment]
I agree with retiring the .pbserver files, I don't see any internally. I was thinking the same thing about the generated RPC client stubs that protobuf puts out that aren't functional. For those, there are one or two subclassed internally, but I was thinking they both could just be replaced by gRPC.

**Review by nichite — APPROVED** [review]
LGTM

**Comment by osa1** [comment]
Why was this PR closed @sigurdm @kevmoo ? Might be good to revive this as it fixes an issue (#485).

**Comment by kevmoo** [comment]
Possible I was doing branch cleanup and didn't realize this was NOT the null safety migration branch.

**Comment by sigurdm** [comment]
> Why was this PR closed @sigurdm @kevmoo ? Might be good to revive this as it fixes an issue.

Yeah good question, we should try and revive it. Not trivial without the branch...

**Comment by sigurdm** [comment]
Reconstructed in https://github.com/google/protobuf.dart/pull/625

**Pull request closed without merging**